### PR TITLE
Adding 1.9 dbt-databricks documentation for new python model config

### PR DESCRIPTION
**Pull request by benc-db**
## What are you changing in this pull request and why?
Adds the first batch of new documentation for dbt-databricks 1.9, focusing on newly support python submission configuration.

## Checklist
- [x] I have reviewed the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [x] The topic I'm writing about is for specific dbt version(s) and I have versioned it according to the [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and/or [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content) guidelines.
- [x] I have added checklist item(s) to this list for anything anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

### CODE REVIEW

File: website/docs/reference/resource-configs/databricks-configs.md [hunk 1]
```diff
@@ -65,6 +65,103 @@ We do not yet have a PySpark API to set tblproperties at table creation, so this
 
 </VersionBlock>
 
+<VersionBlock firstVersion="1.9">
+
+### Python Submission Method
+
+As of 1.9, there are four options for `submission_method`:
```
Comment (amychen1776) [inline website/docs/reference/resource-configs/databricks-configs.md]:
```suggestion
As of 1.9 (or versionless if you're on dbt Cloud), there are four options for `submission_method`: 
```

File: website/docs/reference/resource-configs/databricks-configs.md [hunk 2]
```diff
@@ -65,6 +65,103 @@ We do not yet have a PySpark API to set tblproperties at table creation, so this
...
+| `additional_libs`     | directly configures [libraries](https://docs.databricks.com/api/workspace/jobs/submit#tasks-libraries) | `[]` | ✅ | ✅             | ✅                   | ✅             |
+| `python_job_config`   | additional configuration for jobs/workflows (see table below)        | `{}`               | ✅                     | ✅             | ✅                   | ✅             |
+| `cluster_id`          | id of existing all purpose cluster to execute against                | `None`             | ✅                     | ❌             | ❌                   | ✅             |
+| `http_path`           | path to existing all purpose cluster to execute against              | `None`             | ✅                     | ❌             | ❌                   | ❌             |
+
+\* Only `timeout` and `cluster_id`/`http_path` are supported when `create_notebook` is false
+
+With the introduction of the `workflow_job` submission method we chose to segregate further configuration of the python model submission under a top level configuration named `python_job_config`.
```
Comment (amychen1776) [inline website/docs/reference/resource-configs/databricks-configs.md]:
```suggestion
With the introduction of the `workflow_job` submission method, we chose to segregate further configuration of the python model submission under a top level configuration named `python_job_config`.
```

File: website/docs/reference/resource-configs/databricks-configs.md [hunk 3]
```diff
@@ -65,6 +65,103 @@ We do not yet have a PySpark API to set tblproperties at table creation, so this
 
 </VersionBlock>
 
+<VersionBlock firstVersion="1.9">
+
+### Python submission methods
+
+As of 1.9 (or versionless if you're on dbt Cloud), there are four options for `submission_method`:
```
Comment (runleonarun) [inline website/docs/reference/resource-configs/databricks-configs.md]:
```suggestion
In dbt v1.9 and higher, or in [Versionless](/docs/dbt-versions/versionless-cloud) dbt Cloud, you can use these four options for `submission_method`: 
```

File: website/docs/reference/resource-configs/databricks-configs.md [hunk 4]
```diff
@@ -65,6 +65,103 @@ We do not yet have a PySpark API to set tblproperties at table creation, so this
...
+<VersionBlock firstVersion="1.9">
+
+### Python submission methods
+
+In dbt v1.9 and higher, or in [Versionless](/docs/dbt-versions/versionless-cloud) dbt Cloud, you can use these four options for `submission_method`: 
+
+* `all_purpose_cluster`: execute the python model either directly using the [command api](https://docs.databricks.com/api/workspace/commandexecution) or by uploading a notebook and creating a one-off job run
+* `job_cluster`: creates a new job cluster to execute an uploaded notebook as a one-off job run
+* `serverless_cluster`: uses a [serverless cluster](https://docs.databricks.com/en/jobs/run-serverless-jobs.html) to execute an uploaded notebook as a one-off job run
+* `workflow_job`: creates/updates a reusable workflow and uploaded notebook, for execution on all-purpose, job, or serverless clusters.  :::caution This approach gives you maximum flexibility, but will create persistent artifacts in Databricks (the workflow) that users could run outside of dbt.
```
Comment (runleonarun) [inline website/docs/reference/resource-configs/databricks-configs.md]:
```suggestion
* `workflow_job`: creates/updates a reusable workflow and uploaded notebook, for execution on all-purpose, job, or serverless clusters.  :
::caution 
This approach gives you maximum flexibility, but will create persistent artifacts in Databricks (the workflow) that users could run outside of dbt.
:::
```

File: website/docs/reference/resource-configs/databricks-configs.md [hunk 5]
```diff
@@ -65,6 +65,108 @@ We do not yet have a PySpark API to set tblproperties at table creation, so this
...
+| `existing_job_id`          | Id to use to look up the created workflow (in place of `name`)                                                          | `None`  | ✅             | ❌          |
+| `post_hook_tasks`          | [Tasks](https://docs.databricks.com/api/workspace/jobs/create#tasks) to include after the model notebook execution      | `[]`    | ✅             | ❌          |
+| `additional_task_settings` | Additional [task config])(https://docs.databricks.com/api/workspace/jobs/create#tasks) to include in the model task     | `{}`    | ✅             | ❌          |
+| [Other job run settings](https://docs.databricks.com/api/workspace/jobs/submit) | Config will be copied into the request, outside of the model task  | `None`  | ❌             | ✅          |
+| [Other workflow settings](https://docs.databricks.com/api/workspace/jobs/create) | Config will be copied into the request, outside of the model task | `None`  | ✅             | ❌          |
+
+Here is an example using these new configuration options:
```
Comment (runleonarun) [inline website/docs/reference/resource-configs/databricks-configs.md]:
```suggestion
This example uses the new configuration options in the previous table:
```

File: website/docs/reference/resource-configs/databricks-configs.md [hunk 6]
```diff
@@ -65,6 +65,108 @@ We do not yet have a PySpark API to set tblproperties at table creation, so this
...
+| -------------------------- | ----------------------------------------------------------------------------------------------------------------------- | ------- | -------------- | ---------- |
+| `name`                     | The name to give (or used to look up) the created workflow                                                              | `None`  | ✅             | ❌          |
+| `grants`                   | A simplified way to specify access control for the workflow                                                             | `{}`    | ✅             | ✅          |
+| `existing_job_id`          | Id to use to look up the created workflow (in place of `name`)                                                          | `None`  | ✅             | ❌          |
+| `post_hook_tasks`          | [Tasks](https://docs.databricks.com/api/workspace/jobs/create#tasks) to include after the model notebook execution      | `[]`    | ✅             | ❌          |
+| `additional_task_settings` | Additional [task config])(https://docs.databricks.com/api/workspace/jobs/create#tasks) to include in the model task     | `{}`    | ✅             | ❌          |
```
Comment (runleonarun) [inline website/docs/reference/resource-configs/databricks-configs.md]:
```suggestion
| `additional_task_settings` | Additional [task config](https://docs.databricks.com/api/workspace/jobs/create#tasks) to include in the model task     | `{}`    | ✅             | ❌          |
```

**Review by runleonarun — APPROVED** [review]
Made some minor edits. Approving and merging

**Pull request merged**